### PR TITLE
fix: manual retries

### PR DESCRIPTION
**Pull request by prafulkoppalkar**
## Type of Change

- [x] Bugfix
- [ ] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

Manual Retries not working - intent function was a closure of the `usePaymentIntent` hook 
Hence it was not getting the updated value

## Checklist

- [x] I ran `npm run re:build`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

File: src/Payments/ACHBankDebit.res [hunk 1]
```diff
@@ -76,15 +77,15 @@ let make = (~paymentType: CardThemeType.mode) => {
...
+          intent(~bodyArr=body, ~confirmParam=confirm.confirmParams, ~handleUserError=false,~manualRetry=isManualRetryEnabled, ())
         | None => ()
         }
         ()
       } else {
         postFailedSubmitResponse(~errortype="validation_error", ~message="Please enter all fields")
       }
     }
-  }, (email, modalData, fullName))
+  }, (email, modalData, fullName,isManualRetryEnabled))
```
Comment (vsrivatsa-juspay) [inline src/Payments/ACHBankDebit.res:88]:
```suggestion
  }, (email, modalData, fullName, isManualRetryEnabled))
```

File: src/Payments/ACHBankTransfer.res [hunk 2]
```diff
@@ -32,13 +33,14 @@ let make = (~paymentType: CardThemeType.mode) => {
...
           ~iframeId,
+          ~manualRetry=isManualRetryEnabled,
           (),
         )
       } else {
         postFailedSubmitResponse(~errortype="validation_error", ~message="Please enter all fields")
       }
     }
-  }, [email])
+  }, (email,isManualRetryEnabled))
```
Comment (vsrivatsa-juspay) [inline src/Payments/ACHBankTransfer.res:43]:
```suggestion
  }, (email, isManualRetryEnabled))
```

File: src/Payments/BacsBankTransfer.res [hunk 3]
```diff
@@ -36,13 +37,14 @@ let default = (paymentType: CardThemeType.mode) => {
...
           ~iframeId,
+          ~manualRetry=isManualRetryEnabled,
           (),
         )
       } else {
         postFailedSubmitResponse(~errortype="validation_error", ~message="Please enter all fields")
       }
     }
-  }, [email, fullName])
+  }, (isManualRetryEnabled,email, fullName))
```
Comment (vsrivatsa-juspay) [inline src/Payments/BacsBankTransfer.res:47]:
```suggestion
  }, (isManualRetryEnabled, email, fullName))
```

File: src/Payments/Boleto.res [hunk 4]
```diff
@@ -64,13 +64,14 @@ let make = (~paymentType: CardThemeType.mode) => {
...
           ~iframeId,
+          ~manualRetry=isManualRetryEnabled,
           (),
         )
       } else {
         postFailedSubmitResponse(~errortype="validation_error", ~message="Please enter all fields")
       }
     }
-  }, [socialSecurityNumber])
+  }, (socialSecurityNumber,isManualRetryEnabled))
```
Comment (vsrivatsa-juspay) [inline src/Payments/Boleto.res:74]:
```suggestion
  }, (socialSecurityNumber, isManualRetryEnabled))
```

**Review by ArushKapoorJuspay — APPROVED** [review]

**Review by PritishBudhiraja — APPROVED** [review]
Code wise looks good :')

**Pull request merged**